storage/txcache/testutils_test.go

```diff
@@ -0,0 +1,90 @@
+package txcache
+
+import (
+	"encoding/binary"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+)
+
+const oneMilion = 1000000
+const oneTrilion = oneMilion * oneMilion
+const delta = 0.00000001
+
+func toMicroERD(erd uint64) uint64 {
+	return erd * 1000000
+}
+
+func kBToBytes(kB float32) uint64 {
+	return uint64(kB * 1000)
+}
+
+func (cache *TxCache) getRawScoreOfSender(sender string) float64 {
+	list, ok := cache.txListBySender.getListForSender(sender)
+	if !ok {
+		panic("sender not in cache")
```

Comment:
The panic is only for test-utility code (`getRawScoreOfSender` is only used in tests), to notify the writer of the test that the test setup is not correct.

In production, `getListForSender` is only called from within `addTx` and `removeTx` of the senders map (cache). 

From my analysis, `!ok` happens only on `removeTx`, in this case:

```
func (txMap *txListBySenderMap) onRemoveTxInconsistency(sender string) {
	// This happens when a sender whose transactions were selected for processing is evicted from cache.
	// When it comes to remove one if its transactions due to processing, they don't exist in cache anymore.
	log.Trace("txListBySenderMap.removeTx() detected inconsistency: sender of tx not in cache", "sender", []byte(sender))
}
```

That is, when `cache.RemoveTxByHash` finds and removes the tx in the map of transactions, but when it comes to remove it from the map of senders, it sees that there's no list of transaction associated with this sender - and this happens because of concurrent eviction - according to my understanding - like this:

```
 - transaction of sender is processed
 - transaction removal is triggered (call to cache.RemoveTxByHash)
 - transaction gets removed from tx map (cache.txByHash.removeTx)
 - full capacity triggers eviction and cache.doEvictItems() removes all transactions of the sender and the sender, from both maps
 -  txListBySenderMap.removeTx() calls getListForSender(), which returns !ok (sender already removed)
 - trace message is displayed
```